### PR TITLE
Fix consoletest_setup on ipmi

### PR DESCRIPTION
**Pull request by Zaoliang**
- see https://progress.opensuse.org/issues/3826 for details
- verification run:
  http://e13.suse.de/tests/7980#step/consoletest_setup
- needle PR:
  https://gitlab.suse.de/openqa/os-autoinst-needles-sles/merge_requests/928

### CODE REVIEW

File: lib/utils.pm [hunk 1]
```diff
@@ -266,7 +266,8 @@ sub assert_gui_app {
 # console font, we need to call systemd-vconsole-setup to workaround
 # that
 sub check_console_font {
-    return if check_var('BACKEND', 'spvm');
+    # cover ipmi case now
```
Comment (Soulofdestiny) [inline lib/utils.pm]: can you please rephrase this to something like
`Does not make sense on ssh-based consoles`

File: lib/utils.pm [hunk 2]
```diff
@@ -266,7 +266,8 @@ sub assert_gui_app {
...
+    return if (check_var('BACKEND', 'spvm')) || if (check_var('BACKEND', 'ipmi'));
```
Comment (Soulofdestiny) [inline lib/utils.pm]: the second if is not needed here:
`return if check_var('BACKEND', 'spvm') || check_var('BACKEND', 'ipmi');

File: tests/console/consoletest_setup.pm [hunk 3]
```diff
@@ -16,14 +16,16 @@
...
 
 sub run {
     my $self = shift;
     # let's see how it looks at the beginning
     save_screenshot;
-
-    select_console 'root-console';
+    # use root-ssh for impi
+    check_var("BACKEND", "ipmi") ? use_ssh_serial_console : select_console 'root-console';
+    # select_console 'root-console';
```
Comment (Soulofdestiny) [inline tests/console/consoletest_setup.pm]: please rebase to upstream master first - the change in consoletest_setup is already merged

File: lib/utils.pm (hunk 1 above)
Comment (Zaoliang) [reply lib/utils.pm]: sure, updated, thanks!

**Comment by SergioAtSUSE** [comment]
LGTM

File: tests/console/consoletest_setup.pm [hunk 4]
```diff
@@ -16,14 +16,14 @@
...
+use ipmi_backend_utils 'use_ssh_serial_console';
 use strict;
 
 sub run {
     my $self = shift;
     # let's see how it looks at the beginning
     save_screenshot;
-
-    select_console 'root-console';
+    check_var("BACKEND", "ipmi") ? use_ssh_serial_console : select_console 'root-console';
```
Comment (okurz) [inline tests/console/consoletest_setup.pm]: don't we need the same for other ssh based backends?
